### PR TITLE
chore(CI): add dedupe and upgrade GH actions

### PR DESCRIPTION
**Pull request by jmfrancois**
**What is the problem this PR is trying to solve?**

yarn.lock is never upgraded.
when someone add a dependency it create duplicate

**What is the chosen solution to this problem?**

add two jobs:
* one to upgrade once a week
* one to remove duplicate if someone touch a package.json

do a yarn upgrade to update dependencies to modify the yarn.lock and test the CI

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: .github/workflows/dependencies.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        uses: actions/checkout@v2
+
+      - name: Set up Node
+        uses: actions/setup-node@v1
+        with:
+          node-version: 14
+
+      - name: Upgrade dependencies
+        run: |
+          yarn
```
Comment (frassinier) [inline .github/workflows/dependencies.yml:24]: Using yarn cache setup could be great?

File: .github/workflows/yarn-deduplicate.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+name: yarn-deduplicate
+
+on:
+  pull_request:
+    paths:
+      - 'yarn.lock'
+
+jobs:
+  build:
```
Comment (frassinier) [inline .github/workflows/yarn-deduplicate.yml:9]:
```suggestion
  dedupe:
```

File: .github/workflows/yarn-deduplicate.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        uses: actions/checkout@v2
+
+      - name: yarn-deduplicate
+        id: deduplicate
+        run: |
+          npx yarn-deduplicate
+          yarn
+          echo "::set-output name=diff::$(git status --short yarn.lock)"
+
+      - name: Commit dedup
```
Comment (frassinier) [inline .github/workflows/yarn-deduplicate.yml:24]:
```suggestion
      - name: Commit dedupe
```

**Review by frassinier — APPROVED** [review]
😍

**Review by Kporal — APPROVED** [review]

**Review by romainseb — APPROVED** [review]

**Pull request merged**